### PR TITLE
update docu after stboot redesign

### PR DESCRIPTION
**Pull request by jensdrenhaus**
Signed-off-by: Jens Drenhaus <jens.drenhaus@9elements.com>

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -1,218 +1,284 @@
...
 make check
-```
-
-### Configure stboot target installation
-
-```bash
 # make default configuration
 make config
 # modify configuration
 ${EDITOR} run.config
```
Comment (jensdrenhaus) [inline README.md:28]: Wait for #86 to be done.

File: README.md [hunk 2]
```diff
@@ -1,218 +1,284 @@
...
+    * An Ext4 partition named STDATA containing:
+        * An OS package if boot method is set to _local_.
+        * An empty directory for use as a cache if if boot method is set to _network_.
+
+You need to write the coreboot.rom to the SPI Flash of the host and write the image to the hard drive of the host.
+
+# System Configuration
+
+## Configuration of stboot
+Most options can be set in `run.config`. See the descriptions there for details.
```
Comment (jensdrenhaus) [inline README.md:148]: Wait for #86 to be done.

File: README.md [hunk 3]
```diff
@@ -1,218 +1,284 @@
...
+
+A subset of the configuration options in `run.config` end up in two files whis stboot reads in during the boot process:
```
Comment (jensdrenhaus) [inline README.md:150]: Wait for #86 to be done.

File: README.md [hunk 4]
```diff
@@ -1,218 +1,284 @@
...
-* To enable debug output in _stboot_ pass `-debug`
+If the machine does not have a TPM, stboot will execute the OS package without measuring it. In this case it doesn’t matter if the machine supports Intel TXT or the OS package contains tboot, as Intel TXT requires a TPM.
+
+Note that stboot will always verify the signatures on the OS package before executing it, regardless of what platform security features are available on the machine.
+
+
+# Debugging
+
+## Console output
+The output of stboot can be controlled via the LinuxBoot kernel command line. You can edit the command line in `run.config`. Beside usual kernel parameters you can pass flags to stboot via the special parameter `uroot.uinitargs`.
```
Comment (jensdrenhaus) [inline README.md:280]: Wait for #86 to be done.

File: README.md [hunk 5]
```diff
@@ -221,4 +287,68 @@ Examples:
 
 * print minimal output: `console=ttyS0,115200`
 
-In order to do extensive remote debugging of the host, you can use [u-root's cpu command](DEBUGGING.md).
+## u-root shell
+By setting `ST_LINUXBOOT_VARIANT=full` in `run.config` the LinuxBoot initramfs will contain a shell and u-root's core commands (https://github.com/u-root/u-root/tree/stboot/cmds/core) in addition to stboot itself. So while stboot is running you can press `ctrl+c` to exit. You are then droped into a shell and can inspect the system and use u-roots core commands.
```
Comment (jensdrenhaus) [inline README.md:291]: Wait for #86 to be done.

File: README.md [hunk 6]
```diff
@@ -221,4 +287,68 @@ Examples:
...
+
+## Remote Debugging Using the CPU Command
+In order to do extensive remote debugging of the host, you can use u-root's cpu command. Since the stboot image running on the host has much fewer tools and services than usual Linux operating systems, the `cpu` command is a well suited option for debugging the host remotely.
+It connects to the host, bringing all your local tools and environment with you.
+
+You need to set `ST_LINUXBOOT_VARIANT=debug` in `run.config` in order to include the cpu deamon into the LinuxBoot initramfs.
```
Comment (jensdrenhaus) [inline README.md:297]: Wait for #86 to be done.

File: README.md [hunk 7]
```diff
@@ -1,218 +1,284 @@
...
 
+```
+## OS Package
+In Order to create an example OS package to get started with just do:
 ```bash
-# Generate example sign keys
+# Generate sign keys
 make keygen-sign
+# build configures OS kernel and initramfs, creat and sign an OS package
+make sign
```
Comment (jensdrenhaus) [inline README.md:39]: Wait for #87 to be done.

**Review by jensdrenhaus — COMMENTED** [review]
Wait for #86 and # 87 to be done in order to change the names here.

**Comment by rgdd** [comment]
A pretty huge improvement when compared to the earlier README! =)

Here are some comments from a top-to-bottom read:
- What is the intended target audience? It would be good to state that early on. Right now I think it is "firmware expert". For example, I don't understand the difference between the MBR bootloader and EFI application modes.
- I think the part around host and security configuration is too brief. It needs some additional explaining in addition to just listing the involved fields.
- What is XOR? If it is notation for "or", consider dropping the "X" and just write "or". E.g., "dhcp" or "static".
- "false XOR false" looks odd. Should "use_ospkg_cache" be a boolean?
- I like the feature list. In fact, maybe it should be presented earlier on in one form or another. Right now there are features that the reader sort of have to figure out on their own if they read from top-to-bottom. Network boot is one example.
- Are the figures in doc/ up-to-date? If so, maybe some of them would be a good fit for the README.
- The README is becoming larger. To help the reader navigate efficiently it might be a good idea to provide a roadmap early on.
- There are many typos. Unless someone else takes a pass over the README to fix them I can do it with a PR at some point.

**Comment by jensdrenhaus** [comment]
> A pretty huge improvement when compared to the earlier README! =)
> 
> Here are some comments from a top-to-bottom read:
> 
> * What is the intended target audience? It would be good to state that early on. Right now I think it is "firmware expert". For example, I don't understand the difference between the MBR bootloader and EFI application modes.
> * I think the part around host and security configuration is too brief. It needs some additional explaining in addition to just listing the involved fields.
> * What is XOR? If it is notation for "or", consider dropping the "X" and just write "or". E.g., "dhcp" or "static".
> * "false XOR false" looks odd. Should "use_ospkg_cache" be a boolean?
> * I like the feature list. In fact, maybe it should be presented earlier on in one form or another. Right now there are features that the reader sort of have to figure out on their own if they read from top-to-bottom. Network boot is one example.
> * Are the figures in doc/ up-to-date? If so, maybe some of them would be a good fit for the README.
> * The README is becoming larger. To help the reader navigate efficiently it might be a good idea to provide a roadmap early on.
> * There are many typos. Unless someone else takes a pass over the README to fix them I can do it with a PR at some point.

Re target audience: Very good point to let the reader know at the beginning... OR to further improve the docu. Any suggestions?

Re installation options and configuration files: I'm going to extend the docu.

Re feature list: What would be your suggestion of the order?

Re length of the document. I'll add a TOC once it is nearly ready.

**Comment by jensdrenhaus** [comment]
Merged as part of #115

**Pull request closed without merging**